### PR TITLE
feat: verify signature from event webhook

### PR DESCRIPTION
**Pull request by childish-sambino**
When enabling the "Signed Event Webhook Requests" feature in Mail Settings, Twilio SendGrid will generate a private and public key pair using the Elliptic Curve Digital Signature Algorithm (ECDSA). Once that is successfully enabled, all new event posts will have two new headers: X-Twilio-Email-Event-Webhook-Signature and X-Twilio-Email-Event-Webhook-Timestamp, which can be used to validate your events.

This SDK update will make it easier to verify signatures from signed event webhook requests by using the VerifySignature method. Pass in the public key, event payload, signature, and timestamp to validate. Note: You will need to convert your public key string to an elliptic public key object in order to use the VerifySignature method.

Fixes https://github.com/sendgrid/sendgrid-csharp/issues/1004
Blocked by https://github.com/starkbank/ecdsa-dotnet/issues/4

### CODE REVIEW

File: examples/eventwebhook/RequestValidator.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
+using Microsoft.AspNetCore.Http;
+using SendGrid.Helpers.EventWebbook;
```
Comment (thinkingserious) [inline examples/eventwebhook/RequestValidator.cs:2]: `using SendGrid.Helpers.EventWebbook;` should be `using SendGrid.Helpers.EventWebhook;`

File: src/SendGrid/Helpers/EventWebhook/RequestValidator.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
+﻿using EllipticCurve;
+
+namespace SendGrid.Helpers.EventWebbook
```
Comment (thinkingserious) [inline src/SendGrid/Helpers/EventWebhook/RequestValidator.cs:3]: typo

File: tests/SendGrid.Tests/Helpers/EventWebhook/RequestValidatorTests.cs [hunk 3]
```diff
@@ -0,0 +1,85 @@
+using Xunit;
+using SendGrid.Helpers.EventWebbook;
```
Comment (thinkingserious) [inline tests/SendGrid.Tests/Helpers/EventWebhook/RequestValidatorTests.cs:2]: typo

File: src/SendGrid/Helpers/EventWebhook/RequestValidator.cs [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    /// </summary>
+    public class RequestValidator
+    {
+        /// <summary>
+        /// Signature verification HTTP header name for the signature being sent.
+        /// </summary>
+        public const string SIGNATURE_HEADER = "X-Twilio-Email-Event-Webhook-Signature";
+
+        /// <summary>
+        /// Timestamp http header name for timestamp.
```
Comment (thinkingserious) [inline src/SendGrid/Helpers/EventWebhook/RequestValidator.cs:17]: nit: HTTP

File: src/SendGrid/Helpers/EventWebhook/RequestValidator.cs [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        public const string TIMESTAMP_HEADER = "X-Twilio-Email-Event-Webhook-Timestamp";
+
+        /**
+        * Convert the public key string to a ECPublicKey.
+        *
+        * @param string $publicKey verification key under Mail Settings
+        * @return PublicKey public key using the ECDSA algorithm
+        */
+        /// <summary>
+        /// Convert the public key string to a <see cref="PublicKey"/>ECPublicKey.
```
Comment (thinkingserious) [inline src/SendGrid/Helpers/EventWebhook/RequestValidator.cs:28]: need closing tag?

File: examples/eventwebhook/RequestValidator.cs (hunk 1 above)
Comment (childish-sambino) [reply examples/eventwebhook/RequestValidator.cs]: 🤦

**Comment by ivangalbans** [comment]
hello @childish-sambino ,
Why did you choose https://github.com/starkbank/ecdsa-dotnet instead of working directly with https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.ecdsa?view=netcore-3.1 ?

**Comment by childish-sambino** [comment]
@ivangalbans Simplicity. Very few lines of code to get it working. If you have a working example without this added dependency I can take a look.

**Pull request merged**